### PR TITLE
Java client information.

### PR DESCRIPTION
**Pull request by hayleycd**
#### Summary

Partially addresses #324 by adding Java client information.

#### Release Note
None

### CODE REVIEW

**Comment by hayleycd** [comment]
@loosebazooka tagging you in case you want to review. CC: @haydentherapper @cmurphy

**Comment by loosebazooka** [comment]
I'm not sure how I feel about the duplication of information... These tend to go out of date as the source of truth would be sigstore-java? Should this just be an automated pull in of the readme on sigstore-java (or some translation of it)? Alternatively we could include this is a point to update in our release plan -+ but that does put more work on the devs.

**Comment by hayleycd** [comment]
@loosebazooka that makes a lot of sense. @haydentherapper @cmurphy and I are trying to thread a needle a little bit--a little bit of repetition, but increased exposure, and the devs get to primarily work within their own repositories and aren't forced to standardize between the different language clients. We have open PRs for the other language clients too.

**Comment by hayleycd** [comment]
Thanks for the comments @haydentherapper I have addressed them and merged main.

**Review by haydentherapper — APPROVED** [review]
Approving from my perspective. I'll wait for @loosebazooka to review again. One compromise could be to remove the examples from the docs and point to examples in the repository?

File: content/en/language_clients/java/overview.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+category: Java
+title: Java Client Overview
+weight: 5
+---
+
+[`sigstore-java`](https://github.com/sigstore/sigstore-java#sigstore-java) is a java client for interacting with the Sigstore infrastructure.
+
+## Features
+
+- Includes both [Maven](https://github.com/sigstore/sigstore-java/tree/main/sigstore-maven-plugin) and [Gradle](https://github.com/sigstore/sigstore-java/tree/main/sigstore-gradle) build plugins
```
Comment (loosebazooka) [inline content/en/language_clients/java/overview.md:12]:
```suggestion
- [Maven](https://github.com/sigstore/sigstore-java/tree/main/sigstore-maven-plugin) and [Gradle](https://github.com/sigstore/sigstore-java/tree/main/sigstore-gradle) signing plugins
```

File: content/en/language_clients/java/overview.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+- Keyless signing and verifying
+- [API](https://javadoc.io/doc/dev.sigstore/sigstore-java)
```
Comment (loosebazooka) [inline content/en/language_clients/java/overview.md:14]:
```suggestion
- Java native signing and verifying [API](https://javadoc.io/doc/dev.sigstore/sigstore-java)
```

File: content/en/language_clients/java/overview.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+```java
+plugins {
+    id("dev.sigstore.sign") version "1.0.0"
+}
+```
+
+More information on the Gradle build plugin is available in the [project repository](https://github.com/sigstore/sigstore-java/tree/main/sigstore-gradle#sigstore-gradle).
+
+## Example
```
Comment (loosebazooka) [inline content/en/language_clients/java/overview.md:54]:
```suggestion
## API Usage Examples
```

File: content/en/language_clients/java/overview.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Signing example
```
Comment (loosebazooka) [inline content/en/language_clients/java/overview.md:56]:
```suggestion
### Signing
```

File: content/en/language_clients/java/overview.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+// sign using the sigstore public instance
+var signer = KeylessSigner.builder().sigstorePublicDefaults().build();
+Bundle result = signer.signFile(testArtifact);
+
+// sigstore bundle format (serialized as <artifact>.sigstore.json)
+String bundleJson = result.toJson();
+```
+
+### Verifying example
```
Comment (loosebazooka) [inline content/en/language_clients/java/overview.md:69]:
```suggestion
### Verifying
```

**Review by loosebazooka — COMMENTED** [review]
Just some minor things. I think for most users the gradle and maven plugins are the main entry points. The sigstore-java api is available, but I don't see that as the primary source of adoption.

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**